From b403cd41ad626ec8c20f05aa17db0b57266efde3 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Wed, 25 Sep 2024 13:06:31 +0300
Subject: [PATCH] [FSW-9509] scripts: include compute die 1 mgmt arc

Modified script files to include compute die 1 mgmt arc images
as part of fs1 boot fit file. preboot will load the same image
in all compute dies mgmt arc memories.

Change-Id: I4e8e0c791a97adc9df267b694b0622ab70221d80
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
index 3077315..2088c44 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
@@ -51,6 +51,11 @@
 ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_0_DCCM_LO@/${ARC_MGMT_COMPUTE_0_DCCM_LO}/g;"
 ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_0_ICCM_HI@/${ARC_MGMT_COMPUTE_0_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_0_ICCM_LO@/${ARC_MGMT_COMPUTE_0_ICCM_LO}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_1_DCCM_HI@/${ARC_MGMT_COMPUTE_1_DCCM_HI}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_1_DCCM_LO@/${ARC_MGMT_COMPUTE_1_DCCM_LO}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_1_ICCM_HI@/${ARC_MGMT_COMPUTE_1_ICCM_HI}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_COMPUTE_1_ICCM_LO@/${ARC_MGMT_COMPUTE_1_ICCM_LO}/g;"
+
 
 # Replace ARC core IDs
 ITS_SED_PARAMS+="s/@FIT_CORE_ID_IO0_PID@/${FIT_CORE_ID_IO0_PID}/g;"
diff --git a/board/habanalabs/fs1_arc/fs1-zephyr.its b/board/habanalabs/fs1_arc/fs1-zephyr.its
index 130c239..84498a2 100644
--- a/board/habanalabs/fs1_arc/fs1-zephyr.its
+++ b/board/habanalabs/fs1_arc/fs1-zephyr.its
@@ -66,28 +66,55 @@
 			};
 		};
 
-		zephyr-mgmt_compute-iccm {
-			description = "Zephyr mgmt compute ICCM";
+		zephyr-mgmt_compute_0-iccm {
+			description = "Zephyr mgmt compute 0 ICCM";
 			data = /incbin/("@BINS_DIR_PATH@@DEV_NAME@-zephyr-mgmt_compute-iccm.bin.le");
 			type = "standalone";
 			arch = "arc";
 			core-id = <@FIT_CORE_ID_COMPUTE0_MNG@>;
 			compression = "none";
 			loadaddr = <@ARC_MGMT_COMPUTE_0_ICCM_HI@ @ARC_MGMT_COMPUTE_0_ICCM_LO@>;
-			startaddr = <@MGMT_COMPUTE_STARTADDR_HI@ @MGMT_COMPUTE_STARTADDR_LO@>;
+			startaddr = <@MGMT_COMPUTE_0_STARTADDR_HI@ @MGMT_COMPUTE_0_STARTADDR_LO@>;
 			hash {
 				algo = "@SHA_VERSION@";
 			};
 		};
 
-		zephyr-mgmt_compute-dccm {
-			description = "Zephyr mgmt compute DCCM";
+		zephyr-mgmt_compute_0-dccm {
+			description = "Zephyr mgmt compute 0 DCCM";
 			data = /incbin/("@BINS_DIR_PATH@@DEV_NAME@-zephyr-mgmt_compute-dccm.bin.le");
 			type = "standalone";
 			arch = "arc";
 			core-id = <@FIT_CORE_ID_COMPUTE0_MNG@>;
 			compression = "none";
 			loadaddr = <@ARC_MGMT_COMPUTE_0_DCCM_HI@ @ARC_MGMT_COMPUTE_0_DCCM_LO@>;
+			hash {
+				algo = "@SHA_VERSION@";
+			};
+		};
+
+		zephyr-mgmt_compute_1-iccm {
+			description = "Zephyr mgmt compute 1 ICCM";
+			data = /incbin/("@BINS_DIR_PATH@@DEV_NAME@-zephyr-mgmt_compute-iccm.bin.le");
+			type = "standalone";
+			arch = "arc";
+			core-id = <@FIT_CORE_ID_COMPUTE0_MNG@>;
+			compression = "none";
+			loadaddr = <@ARC_MGMT_COMPUTE_1_ICCM_HI@ @ARC_MGMT_COMPUTE_1_ICCM_LO@>;
+			startaddr = <@MGMT_COMPUTE_1_STARTADDR_HI@ @MGMT_COMPUTE_1_STARTADDR_LO@>;
+			hash {
+				algo = "@SHA_VERSION@";
+			};
+		};
+
+		zephyr-mgmt_compute_1-dccm {
+			description = "Zephyr mgmt compute 1 DCCM";
+			data = /incbin/("@BINS_DIR_PATH@@DEV_NAME@-zephyr-mgmt_compute-dccm.bin.le");
+			type = "standalone";
+			arch = "arc";
+			core-id = <@FIT_CORE_ID_COMPUTE0_MNG@>;
+			compression = "none";
+			loadaddr = <@ARC_MGMT_COMPUTE_1_DCCM_HI@ @ARC_MGMT_COMPUTE_1_DCCM_LO@>;
 			hash {
 				algo = "@SHA_VERSION@";
 			};
@@ -101,7 +128,8 @@
 			load-type = <0>; /* FIT_LOAD_TYPE_RUN */
 			description = "Habanalabs device preboot";
 			loadables = /* "zephyr-pid-iccm", "zephyr-pid-dccm", */ "zephyr-mgmt-iccm",
-				    "zephyr-mgmt-dccm", "zephyr-mgmt_compute-iccm", "zephyr-mgmt_compute-dccm";
+				    "zephyr-mgmt-dccm", "zephyr-mgmt_compute_0-iccm", "zephyr-mgmt_compute_0-dccm",
+					"zephyr-mgmt_compute_1-iccm", "zephyr-mgmt_compute_1-dccm";
 		};
 	};
 };
diff --git a/board/habanalabs/fs1_arc/post-image-fit.sh b/board/habanalabs/fs1_arc/post-image-fit.sh
index c573199..6884b60 100755
--- a/board/habanalabs/fs1_arc/post-image-fit.sh
+++ b/board/habanalabs/fs1_arc/post-image-fit.sh
@@ -57,12 +57,14 @@
 # is same as ARC_XX_ICCM_HI.
 PID_STARTADDR_LO=$((ARC_PID_ICCM_LO | PID_STARTADDR))
 MGMT_STARTADDR_LO=$((ARC_MGMT_ICCM_LO | MGMT_STARTADDR))
-MGMT_COMPUTE_STARTADDR_LO=$((ARC_MGMT_COMPUTE_0_ICCM_LO | MGMT_COMPUTE_STARTADDR))
+MGMT_COMPUTE_0_STARTADDR_LO=$((ARC_MGMT_COMPUTE_0_ICCM_LO | MGMT_COMPUTE_STARTADDR))
+MGMT_COMPUTE_1_STARTADDR_LO=$((ARC_MGMT_COMPUTE_1_ICCM_LO | MGMT_COMPUTE_STARTADDR))
 
 ITS_SED_PARAMS+="s/@PID_STARTADDR_HI@/${ARC_PID_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@PID_STARTADDR_LO@/${PID_STARTADDR_LO}/g;"
 ITS_SED_PARAMS+="s/@MGMT_STARTADDR_HI@/${ARC_MGMT_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@MGMT_STARTADDR_LO@/${MGMT_STARTADDR_LO}/g;"
-ITS_SED_PARAMS+="s/@MGMT_COMPUTE_STARTADDR_HI@/${ARC_MGMT_COMPUTE_0_ICCM_HI}/g;"
-ITS_SED_PARAMS+="s/@MGMT_COMPUTE_STARTADDR_LO@/${MGMT_COMPUTE_STARTADDR_LO}/g;"
-
+ITS_SED_PARAMS+="s/@MGMT_COMPUTE_0_STARTADDR_HI@/${ARC_MGMT_COMPUTE_0_ICCM_HI}/g;"
+ITS_SED_PARAMS+="s/@MGMT_COMPUTE_0_STARTADDR_LO@/${MGMT_COMPUTE_0_STARTADDR_LO}/g;"
+ITS_SED_PARAMS+="s/@MGMT_COMPUTE_1_STARTADDR_HI@/${ARC_MGMT_COMPUTE_1_ICCM_HI}/g;"
+ITS_SED_PARAMS+="s/@MGMT_COMPUTE_1_STARTADDR_LO@/${MGMT_COMPUTE_1_STARTADDR_LO}/g;"
